plugins/modules/packaging/os/pacman.py

```diff
@@ -254,16 +254,23 @@ def upgrade(module, pacman_path):
         # e.g., "ansible 2.7.1-1 -> 2.7.2-1"
         regex = re.compile(r'([\w+\-.@]+) (\S+-\S+) -> (\S+-\S+)')
         for p in data:
-            m = regex.search(p)
-            packages.append(m.group(1))
-            if module._diff:
-                diff['before'] += "%s-%s\n" % (m.group(1), m.group(2))
-                diff['after'] += "%s-%s\n" % (m.group(1), m.group(3))
+            if 'ignored' not in p:
```

Comment:
What happens if `ignored` is part of a package name?

Reply:
Nice catch! Fixed in https://github.com/ansible-collections/community.general/pull/2936/commits/d55b0c1336b6be5bfcea7feada5b60f51b9f95d8.